garminexport/incremental_backup.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/env python
+import getpass
+import logging
+import os
+from datetime import timedelta
+
+import garminexport.backup
+from garminexport.backup import export_formats
+from garminexport.garminclient import GarminClient
+from garminexport.logging_config import LOG_LEVELS
+from garminexport.retryer import Retryer, ExponentialBackoffDelayStrategy, MaxRetriesStopStrategy
+
+log = logging.getLogger(__name__)
+
+
+def incremental_backup(username: str,
+                       password: str = None,
+                       backup_dir: str = os.path.join(".", "activities"),
+                       format: str = 'ALL',
+                       log_level: str = 'INFO',
+                       ignore_errors: bool = False,
+                       max_retries: int = 7):
+    """Performs (incremental) backups of activities for a given Garmin Connect account.
+
+    :param username: Garmin Connect user name
+    :param password: Garmin Connect user password. Default: None. If not provided, would be asked interactively.
+    :param backup_dir: Destination directory for downloaded activities. Default: ./activities/".
+    :param format: Desired output formats (json_summary, json_details, gpx, tcx, fit). Default: ALL.
+    :param log_level: Desired log output level (DEBUG, INFO, WARNING, ERROR). Default: INFO.
+    :param ignore_errors: Ignore errors and keep going. Default: False.
+    :param max_retries: The maximum number of retries to make on failed attempts to fetch an activity.
+    Exponential backoff will be used, meaning that the delay between successive attempts
+    will double with every retry, starting at one second. Default: 7.
+
+    The activities are stored in a local directory on the user's computer.
+    The backups are incremental, meaning that only activities that aren't already
+    stored in the backup directory will be downloaded.
+    """
+    if log_level not in LOG_LEVELS:
+        raise ValueError("Illegal log-level: {}".format(log_level))
```

Comment:
We can drop this check without the `log_level` argument.

Reply:
removed